x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferencePlugin.java

```diff
@@ -274,7 +277,12 @@ public Collection<?> createComponents(PluginServices services) {
 
             ElasticInferenceServiceSettings inferenceServiceSettings = new ElasticInferenceServiceSettings(settings);
             String elasticInferenceUrl = this.getElasticInferenceServiceUrl(inferenceServiceSettings);
-            elasticInferenceServiceComponents.set(new ElasticInferenceServiceComponents(elasticInferenceUrl));
+            elasticInferenceServiceComponents.set(
```

Comment:
@brendan-jugan-elastic this is where we'll need the logic to retrieve the actual enabled models and task types from the EIS gateway.